File: gamebet_backend_python/account/account_store.py
import time
from esport_api import store as main_store

FILES = {
    "tagPlatforms": "tag_platforms",
    "players": "players",
    "moneyLogs": "money_logs",
    "playerOrders": "player_orders",
}


def _now_ms() -> int:
    return int(time.time() * 1000)


def _next_id(collection: dict) -> int:
    max_id = 0
    for row in collection.values():
        rid = int(row.get("id") or row.get("ID") or row.get("playerId") or 0)
        if rid > max_id:
            max_id = rid
    return max_id + 1


def ensure_seed():
    for key in FILES.values():
        if main_store.read_json(key, None) is None:
            main_store.write_json(key, {})


def list_tag_platforms() -> list:
    all_p = main_store.read_json(FILES["tagPlatforms"], {})
    return sorted([{"ID": r["id"], "Name": r["name"]} for r in all_p.values()], key=lambda x: x["ID"])


def find_tag_platform_by_name(name: str):
    all_p = main_store.read_json(FILES["tagPlatforms"], {})
    return next((r for r in all_p.values() if r.get("name") == name), None)


def create_tag_platform(platform_name: str, player_name: str) -> dict:
    platforms = main_store.read_json(FILES["tagPlatforms"], {})
    players = main_store.read_json(FILES["players"], {})

    platform = find_tag_platform_by_name(platform_name)
    if not platform:
        pid = _next_id(platforms)
        platform = {"id": pid, "name": platform_name, "createdAt": _now_ms()}
        platforms[str(pid)] = platform

    player_id = _next_id(players)
    player = {
        "id": player_id, "platformId": platform["id"], "platformName": platform["name"],
        "playerName": player_name or "", "credit": 0, "totalBalance": 0,
        "createdAt": _now_ms(), "updatedAt": _now_ms(),
    }
    players[str(player_id)] = player

    main_store.write_json(FILES["tagPlatforms"], platforms)
    main_store.write_json(FILES["players"], players)

    return {"playerId": player_id, "playerName": player["playerName"],
            "platformId": platform["id"], "platformName": platform["name"]}


def get_player(player_id) -> dict | None:
    players = main_store.read_json(FILES["players"], {})
    return players.get(str(player_id))


def update_player_balance(player_id, balance: float) -> dict | None:
    players = main_store.read_json(FILES["players"], {})
    key = str(player_id)
    row = players.get(key)
    if not row:
        return None
    row["totalBalance"] = float(balance)
    row["updatedAt"] = _now_ms()
    players[key] = row
    main_store.write_json(FILES["players"], players)
    return row


def delete_player(player_id, description: str = "") -> bool:
    players = main_store.read_json(FILES["players"], {})
    key = str(player_id)
    if key not in players:
        return False
    del players[key]
    main_store.write_json(FILES["players"], players)
    return True


def list_money_logs(player_id, page_index: int = 1, page_size: int = 20) -> dict:
    all_logs = main_store.read_json(FILES["moneyLogs"], {})
    logs = sorted(
        [v for v in all_logs.values() if str(v.get("playerId")) == str(player_id)],
        key=lambda x: x.get("createAt", 0), reverse=True,
    )
    total = len(logs)
    start = (page_index - 1) * page_size
    return {"data": logs[start:start + page_size], "total": total,
            "pageIndex": page_index, "pageSize": page_size}


def get_money_log(log_id) -> dict | None:
    return main_store.read_json(FILES["moneyLogs"], {}).get(str(log_id))


def save_money_log(body: dict) -> dict:
    all_logs = main_store.read_json(FILES["moneyLogs"], {})
    log_id = body.get("logId") or _next_id(all_logs)
    row = {**body, "logId": log_id, "createAt": body.get("createAt") or _now_ms()}
    all_logs[str(log_id)] = row
    main_store.write_json(FILES["moneyLogs"], all_logs)
    return row


def delete_money_log(log_id) -> bool:
    all_logs = main_store.read_json(FILES["moneyLogs"], {})
    key = str(log_id)
    if key not in all_logs:
        return False
    del all_logs[key]
    main_store.write_json(FILES["moneyLogs"], all_logs)
    return True


def save_player_orders(player_id, order_type: str, orders: list):
    all_orders = main_store.read_json(FILES["playerOrders"], {})
    all_orders[str(player_id)] = {"playerId": player_id, "type": order_type,
                                   "orders": orders, "savedAt": _now_ms()}
    main_store.write_json(FILES["playerOrders"], all_orders)


def list_by_player(player_id) -> list:
    all_orders = main_store.read_json(FILES["playerOrders"], {})
    block = all_orders.get(str(player_id))
    return block.get("orders", []) if block else []


def get_accounts_from_kv() -> list:
    raw = main_store.get_user_kv("ACCOUNT")
    if not raw:
        return []
    try:
        import json
        return json.loads(raw) if isinstance(raw, str) else (raw if isinstance(raw, list) else [])
    except Exception:
        return []

File: changmen/gamebet_backend/core/account/account_store.js
"use strict";

const store = require("../esport-api/store.js");

const FILES = {
  tagPlatforms: "tag_platforms",
  players: "players",
  moneyLogs: "money_logs",
  playerOrders: "player_orders",
};

function nextId(collection) {
  let max = 0;
  for (const row of Object.values(collection || {})) {
    const id = Number(row.id ?? row.ID ?? row.playerId ?? 0);
    if (id > max) max = id;
  }
  return max + 1;
}

function listTagPlatforms() {
  const all = store.readJson(FILES.tagPlatforms, {});
  return Object.values(all)
    .map((row) => ({ ID: row.id, Name: row.name }))
    .sort((a, b) => a.ID - b.ID);
}

function findTagPlatformByName(name) {
  const all = store.readJson(FILES.tagPlatforms, {});
  return Object.values(all).find((row) => row.name === name) || null;
}

function createTagPlatform(platformName, playerName) {
  const platforms = store.readJson(FILES.tagPlatforms, {});
  const players = store.readJson(FILES.players, {});

  let platform = findTagPlatformByName(platformName);
  if (!platform) {
    const id = nextId(platforms);
    platform = { id, name: platformName, createdAt: Date.now() };
    platforms[id] = platform;
  }

  const playerId = nextId(players);
  const player = {
    id: playerId,
    platformId: platform.id,
    platformName: platform.name,
    playerName: playerName || "",
    credit: 0,
    totalBalance: 0,
    createdAt: Date.now(),
    updatedAt: Date.now(),
  };
  players[playerId] = player;

  store.writeJson(FILES.tagPlatforms, platforms);
  store.writeJson(FILES.players, players);

  return {
    playerId,
    playerName: player.playerName,
    platformId: platform.id,
    platformName: platform.name,
  };
}

function getPlayer(playerId) {
  const players = store.readJson(FILES.players, {});
  return players[String(playerId)] || null;
}

function updatePlayerBalance(playerId, balance) {
  const players = store.readJson(FILES.players, {});
  const key = String(playerId);
  const row = players[key];
  if (!row) return null;

  row.totalBalance = Number(balance) || 0;
  row.updatedAt = Date.now();
  players[key] = row;
  store.writeJson(FILES.players, players);

  const platforms = store.readJson(FILES.tagPlatforms, {});
  const platform = platforms[String(row.platformId)];

  return {
    total: row.totalBalance,
    platformId: row.platformId,
    platformName: platform?.name || row.platformName || "",
    credit: row.credit || 0,
  };
}

function deletePlayer(playerId, description) {
  const players = store.readJson(FILES.players, {});
  const key = String(playerId);
  if (!players[key]) return false;

  players[key] = {
    ...players[key],
    deletedAt: Date.now(),
    deleteDescription: description || "",
  };
  store.writeJson(FILES.players, players);

  const orders = store.readJson(FILES.playerOrders, {});
  delete orders[key];
  store.writeJson(FILES.playerOrders, orders);

  const logs = store.readJson(FILES.moneyLogs, {});
  for (const [logId, row] of Object.entries(logs)) {
    if (String(row.playerId) === key) delete logs[logId];
  }
  store.writeJson(FILES.moneyLogs, logs);

  removeAccountFromKv(key);
  return true;
}

function removeAccountFromKv() {}

function listMoneyLogs(playerId, pageIndex = 1, pageSize = 20) {
  const all = store.readJson(FILES.moneyLogs, {});
  const list = Object.values(all)
    .filter((row) => String(row.playerId) === String(playerId))
    .sort((a, b) => (b.createAt || 0) - (a.createAt || 0));

  const start = (pageIndex - 1) * pageSize;
  return {
    list: list.slice(start, start + pageSize),
    data: list,
    total: list.length,
    RecordCount: list.length,
    pageIndex,
    pageSize,
  };
}

function getMoneyLog(logId) {
  const all = store.readJson(FILES.moneyLogs, {});
  return all[String(logId)] || null;
}

function saveMoneyLog(payload) {
  const all = store.readJson(FILES.moneyLogs, {});
  const logId = payload.logId || nextId(all);
  const row = {
    logId,
    playerId: payload.playerId,
    type: payload.type || "Recharge",
    money: Number(payload.money) || 0,
    description: payload.description || "",
    createAt: payload.createAt || Date.now(),
    updatedAt: Date.now(),
  };
  all[String(logId)] = row;
  store.writeJson(FILES.moneyLogs, all);
  return row;
}

function deleteMoneyLog(logId) {
  const all = store.readJson(FILES.moneyLogs, {});
  if (!all[String(logId)]) return false;
  delete all[String(logId)];
  store.writeJson(FILES.moneyLogs, all);
  return true;
}

function getPlayerOrders(playerId) {
  const all = store.readJson(FILES.playerOrders, {});
  return all[String(playerId)]?.orders || [];
}

function savePlayerOrders(playerId, provider, orders) {
  const all = store.readJson(FILES.playerOrders, {});
  const key = String(playerId);
  const prev = all[key]?.orders || [];
  const merged = [...prev];

  for (const order of orders || []) {
    const idx = merged.findIndex(
      (row) => row.orderId === order.orderId && row.provider === (order.provider || provider)
    );
    if (idx >= 0) merged[idx] = { ...merged[idx], ...order };
    else merged.push({ ...order, provider: order.provider || provider });
  }

  all[key] = { playerId: key, orders: merged, updatedAt: Date.now() };
  store.writeJson(FILES.playerOrders, all);
  return merged;
}

function getAccountsFromKv() { return []; }

function ensureSeed() {
  store.ensureSeed();
  for (const name of Object.values(FILES)) {
    if (store.readJson(name, null) == null) store.writeJson(name, {});
  }
}

module.exports = {
  FILES,
  ensureSeed,
  listTagPlatforms,
  createTagPlatform,
  getPlayer,
  updatePlayerBalance,
  deletePlayer,
  listMoneyLogs,
  getMoneyLog,
  saveMoneyLog,
  deleteMoneyLog,
  getPlayerOrders,
  savePlayerOrders,
  getAccountsFromKv,
  removeAccountFromKv,
};
